editor/plugins/texture_editor_plugin.cpp

```diff
@@ -29,133 +29,62 @@
 /*************************************************************************/
 
 #include "texture_editor_plugin.h"
+#include "editor/editor_scale.h"
 
-#include "core/config/project_settings.h"
-#include "core/io/resource_loader.h"
-#include "editor/editor_settings.h"
+TexturePreview::TexturePreview(Ref<Texture2D> p_texture, bool p_show_metadata) {
+	Ref<Theme> theme = EditorNode::get_singleton()->get_editor_theme();
 
-void TextureEditor::_gui_input(Ref<InputEvent> p_event) {
-}
-
-void TextureEditor::_notification(int p_what) {
-	if (p_what == NOTIFICATION_READY) {
-		//get_scene()->connect("node_removed",this,"_node_removed");
-	}
-
-	if (p_what == NOTIFICATION_DRAW) {
-		Ref<Texture2D> checkerboard = get_theme_icon("Checkerboard", "EditorIcons");
-		Size2 size = get_size();
+	TextureRect *checkerboard = memnew(TextureRect);
+	checkerboard->set_texture(theme->get_icon("Checkerboard", "EditorIcons"));
+	checkerboard->set_stretch_mode(TextureRect::STRETCH_TILE);
+	checkerboard->set_custom_minimum_size(Size2(0.0, 256.0) * EDSCALE);
```

Comment:
Adding `checkerboard->set_texture_repeat(CanvasItem::TEXTURE_REPEAT_ENABLED);` will fix the weird stretching bug.

Reply:
This didn't happen when I tested on 3.x (my laptop doesn't support vulkan). It seems a little weird to me that setting the stretch mode to tile wouldn't actually cause it to tile, should `CanvasItem::set_texture_repeat()` maybe be called from `TextureRect::set_stretch_mode()`? Or maybe `TextureRect::STRETCH_TILE` should just be removed altogether in favor of using `CanvasItem::TEXTURE_REPEAT_ENABLED` + `TexureRect::STRETCH_SCALE` (does that even work?)

Reply:
STRETCH_SCALE is something completely different, we can't have a flag that has a different behavior based on texture settings. It's counter-intuitive.

Although the fact that STRETCH_TILE doesn't work without repeat sounds like a bug 🤔 but it makes sense, old code probably used some custom drawing to make sure it works with any texture and that's less efficient.